src/wp-includes/l10n/class-wp-translation-file-mo.php

```diff
@@ -0,0 +1,219 @@
+		$offsets['translations_length'] = $offsets['hash_addr'] - $offsets['translations_addr'];
+
+		if ( $offsets['rev'] > 0 ) {
+			$this->error = 'Unsupported Revision.';
+			return false;
+		}
+
+		if ( $offsets['translations_addr'] > $file_length || $offsets['originals_addr'] > $file_length ) {
+			$this->error = 'Invalid Data.';
+			return false;
+		}
+
+		// Load the Originals.
+		$original_data     = str_split( substr( $file_contents, $offsets['originals_addr'], $offsets['originals_length'] ), 8 );
+		$translations_data = str_split( substr( $file_contents, $offsets['translations_addr'], $offsets['translations_length'] ), 8 );
+
+		foreach ( array_keys( $original_data ) as $i ) {
+			$o = unpack( "{$this->uint32}length/{$this->uint32}pos", $original_data[ $i ] );
+			$t = unpack( "{$this->uint32}length/{$this->uint32}pos", $translations_data[ $i ] );
+
+			if ( false === $o || false === $t ) {
+				continue;
+			}
+
+			$original    = substr( $file_contents, $o['pos'], $o['length'] );
+			$translation = substr( $file_contents, $t['pos'], $t['length'] );
+			// GlotPress bug.
+			$translation = rtrim( $translation, "\0" );
+
+			// Metadata about the MO file is stored in the first translation entry.
+			if ( '' === $original ) {
+				foreach ( explode( "\n", $translation ) as $meta_line ) {
+					if ( '' === $meta_line ) {
+						continue;
+					}
+
+					list( $name, $value ) = array_map( 'trim', explode( ':', $meta_line, 2 ) );
+
+					$this->headers[ strtolower( $name ) ] = $value;
+				}
+			} else {
+				$this->entries[ (string) $original ] = $translation;
+			}
+		}
+
+		return true;
+	}
+
+	/**
+	 * Exports translation contents as a string.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @return string Translation file contents.
+	 */
+	public function export() {
+		// Prefix the headers as the first key.
+		$headers_string = '';
+		foreach ( $this->headers as $header => $value ) {
+			$headers_string .= "{$header}: $value\n";
+		}
+		$entries     = array_merge( array( '' => $headers_string ), $this->entries );
+		$entry_count = count( $entries );
+
+		if ( false === $this->uint32 ) {
+			$this->uint32 = 'V';
+		}
+
+		$bytes_for_entries = $entry_count * 4 * 2;
+		// Pair of 32bit ints per entry.
+		$originals_addr    = 28; /* header */
+		$translations_addr = $originals_addr + $bytes_for_entries;
+		$hash_addr         = $translations_addr + $bytes_for_entries;
+		$entry_offsets     = $hash_addr;
+
+		$file_header = pack( $this->uint32 . '*', self::MAGIC_MARKER, 0 /* rev */, $entry_count, $originals_addr, $translations_addr, 0 /* hash_length */, $hash_addr );
+
+		$o_entries = '';
+		$t_entries = '';
+		$o_addr    = '';
+		$t_addr    = '';
+
+		foreach ( array_keys( $entries ) as $original ) {
+			$o_addr        .= pack( $this->uint32 . '*', strlen( $original ), $entry_offsets );
+			$entry_offsets += strlen( $original ) + 1;
+			$o_entries     .= $original . pack( 'x' );
```

Comment:
PhpStorm is warning about this lacking the `$values` parameter.

![image](https://github.com/WordPress/wordpress-develop/assets/134745/78dc0b06-6f01-43a3-b024-e6d4088a1974)

Is this being used just to obtain a null byte? If so, should this instead be better as:

```suggestion
			$o_entries     .= $original . "\0";
```